runtime/js/41_prompt.js

```diff
@@ -1,79 +1,142 @@
 import { stdin } from "ext:deno_io/12_io.js";
-const { ArrayPrototypePush, StringPrototypeCharCodeAt, Uint8Array } =
-  primordials;
-const LF = StringPrototypeCharCodeAt("\n", 0);
-const CR = StringPrototypeCharCodeAt("\r", 0);
+import { bold, cyan, italic, yellow } from "ext:deno_console/01_colors.js";
+const {
+  Array,
+  ArrayPrototypeFill,
+  ArrayPrototypeJoin,
+  ArrayPrototypeMap,
+  Error,
+  Map,
+  Uint8Array,
+  StringFromCodePoint,
+} = primordials;
+
+const LINE_UP = "\x1b[1A";
+const LINE_CLEAR = "\x1b[2K";
+const HIDE_CURSOR = "\x1b[?25l";
+const SHOW_CURSOR = "\x1b[?25h";
+const UP_ARROW = "\x1b[A";
+const DOWN_ARROW = "\x1b[B";
+const ESC = "\x1b";
+const CTRL_C = "\x03";
+const CTRL_D = "\x04";
 
 function alert(message = "Alert") {
   if (!isatty(stdin.rid)) {
-    return;
+    throw new Error("Cannot use `alert` in a non-interactive environment");
   }
 
-  core.print(`${message} [Enter] `, false);
+  core.print(
+    `${yellow(bold(`${message}`))} [${italic("Press any key to continue")}] `,
+  );
 
-  readLineFromStdinSync();
+  try {
+    stdin.setRaw(true);
+    stdin.readSync(new Uint8Array(1024));
+  } finally {
+    stdin.setRaw(false);
+  }
+
+  core.print("\n");
 }
 
-function confirm(message = "Confirm") {
+function prompt(message = "Prompt", defaultValue = "") {
   if (!isatty(stdin.rid)) {
-    return false;
+    throw new Error("Cannot use `prompt` in a non-interactive environment");
   }
 
-  core.print(`${message} [y/N] `, false);
-
-  const answer = readLineFromStdinSync();
-
-  return answer === "Y" || answer === "y";
+  return ops.op_read_line_prompt(
+    `${yellow(bold(`${message}`))} `,
+    `${defaultValue}`,
+  );
 }
 
-function prompt(message = "Prompt", defaultValue) {
-  defaultValue ??= null;
-
+function confirm(message = "Confirm") {
   if (!isatty(stdin.rid)) {
-    return null;
+    throw new Error("Cannot use `confirm` in a non-interactive environment");
   }
 
-  if (defaultValue) {
-    message += ` [${defaultValue}]`;
-  }
-
-  message += " ";
+  const options = [{ val: true, desc: "OK" }, { val: false, desc: "Cancel" }];
+  const inputMap = new Map([
+    ["y", true],
+    ["n", false],
+    ["Y", true],
+    ["N", false],
+    [ESC, false],
+    [CTRL_C, false],
+    [CTRL_D, false],
+  ]);
+
+  core.print(`${yellow(bold(`${message}`))}\n`);
+  return select(options, inputMap);
+}
 
-  // output in one shot to make the tests more reliable
-  core.print(message, false);
+function select(options, inputMap) {
+  let val = options.at(-1).val;
+  try {
+    core.print(HIDE_CURSOR);
+    stdin.setRaw(true);
+    val = selectOption(options, inputMap);
+  } finally {
+    core.print(SHOW_CURSOR);
+    stdin.setRaw(false);
+  }
 
-  return readLineFromStdinSync() || defaultValue;
+  core.print("\n");
+  return val;
 }
 
-function readLineFromStdinSync() {
-  const c = new Uint8Array(1);
-  const buf = [];
+function selectOption(options, inputMap) {
+  const selectedIdx = 0;
```

Comment:
Looks reassigned?

```suggestion
  let selectedIdx = 0;
```

Reply:
Correct - fixed locally but forgot to push